server/test/repository/ApplicationStatusesRepositoryTest.java

```diff
@@ -85,4 +89,31 @@ public void canQueryForOboseleteApplicationStatuses() {
     assertThat(statusDefinitionsResults.get(0).getStatuses().get(0).statusText())
         .isEqualTo("Approved");
   }
+
+  @Test
+  public void canUpdateApplicationStatuses() {
+    Long uniqueProgramId = new Random().nextLong();
+    ProgramModel program =
+        ProgramBuilder.newActiveProgram("Updateprogram" + uniqueProgramId, "description").build();
+    String programName = program.getProgramDefinition().adminName();
+    StatusDefinitions statusDefinitions = new StatusDefinitions(ImmutableList.of(APPROVED_STATUS));
+    ApplicationStatusesModel applicationStatusesModel =
+        new ApplicationStatusesModel(
+            programName, statusDefinitions, StatusDefinitionsLifecycleStage.ACTIVE);
+    applicationStatusesModel.save();
+    StatusDefinitions statusDefinitionsResult = repo.lookupActiveStatusDefinitions(programName);
+    assertStatusDefinitions(statusDefinitionsResult, "Approved");
+
+    repo.updateStatusDefinitions(
+        programName, new StatusDefinitions(ImmutableList.of(REAPPLY_STATUS)));
+
+    StatusDefinitions statusDefinitionsResult2 = repo.lookupActiveStatusDefinitions(programName);
+    assertStatusDefinitions(statusDefinitionsResult2, "Reapply");
+  }
+
+  public void assertStatusDefinitions(
+      StatusDefinitions statusDefinitionsResult, String statusText) {
+    assertThat(statusDefinitionsResult.getStatuses().size()).isEqualTo(1);
+    assertThat(statusDefinitionsResult.getStatuses().get(0).statusText()).isEqualTo(statusText);
+  }
```

Comment:
**suggestion, optional, not-blocking**: Tests don't need to follow DRY principles as much as regular code, since the primary goal is readability. imo leaving these asserts in each test, even if the code is repetitive, is better cause it makes it easier to see what the test is doing.

Reply:
I agree with you! Let me move it back into each tests